generators/entity/templates/src/main/java/package/domain/_Entity.java

```diff
@@ -142,6 +144,9 @@
 
     <%_ } else if (relationshipType == 'many-to-one') { _%>
     @ManyToOne
+    <%_ if (relationshipValidate == true && relationshipType == 'many-to-one' && relationshipValidateRules.indexOf('required') != -1) { _%>
```

Comment:
I think you dont have to check the relationship types again as you will set `relationshipValidate ` only when thse conditions are met so you can simplify to `<%_ if (relationshipValidate) { %>`